### ISSUE TITLE
Can't get to work on Alpine

### ISSUE THREAD

**Issue by aknuds1** [aerospike/aerospike-client-nodejs#286]
I'm trying to install aerospike-client-nodejs on Alpine Linux, using C library built from source, but it fails due to a mysterious relocation error. Please help me figure this one out!

The below Dockerfile reproduces the error (given a package.json file in the current directory):

```
FROM node:10-alpine
MAINTAINER Me

WORKDIR /app

ENV PREFIX=/opt/aerospike-client-c

RUN apk add --no-cache python2 make gcc g++ git libtool autoconf automake linux-headers \
openssl-dev zlib-dev libuv-dev

RUN git clone https://github.com/aerospike/aerospike-client-c.git && cd aerospike-client-c && \
git submodule update --init && \
make EVENT_LIB=libuv && \
mv target/* /opt/aerospike-client-c

COPY package.json .

RUN npm install aerospike
RUN node -e "require('aerospike')"
```

You should see an error like

```
/app/node_modules/bindings/bindings.js:88
        throw e
        ^

Error: Error relocating /app/node_modules/aerospike/build/Release/aerospike.node: sk_free: symbol not found
    at Object.Module._extensions..node (internal/modules/cjs/loader.js:717:18)
    at Module.load (internal/modules/cjs/loader.js:598:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:537:12)
    at Function.Module._load (internal/modules/cjs/loader.js:529:3)
    at Module.require (internal/modules/cjs/loader.js:636:17)
    at require (internal/modules/cjs/helpers.js:20:18)
    at bindings (/app/node_modules/bindings/bindings.js:81:44)
    at Object.<anonymous> (/app/node_modules/aerospike/lib/aerospike.js:19:31)
    at Module._compile (internal/modules/cjs/loader.js:688:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:699:10)
```

**Comment by jhecking** [comment]
The client used to build/run on Alpine Linux in the past. However, I can reproduce the issue using your Dockerfile. There are at least two issues that are currently causing the build to fail:

(1) In `binding.gyp` I need to add `-lssl` to link against libssl explicitly.
(2) The Aerospike C client recently added a call to `pthread_attr_setaffinity_np` - this function does not seem to be supported on Alpine Linux.

I'll fix (1) but I'll need to talk to our C client developer re (2).

I've temporarily patched both issues in my docker image and am able to run the client.

**Comment by jhecking** [comment]
> How do you patch (2) though?

By patching the as_cpu.h header file in the Aerospike C client to remove the problematic function call.

**Comment by aknuds1** [comment]
@jhecking I made [a PR](https://github.com/aerospike/aerospike-client-nodejs/pull/288) for the ssl linking.

Can you explain exactly how you patched as_cpu.h? Did you just avoid calling `pthread_attr_setaffinity_np` (and return 0 instead)?

**Comment by jhecking** [comment]
> Did you just avoid calling pthread_attr_setaffinity_np (and return 0 instead)?

Exactly.

**Comment by aknuds1** [comment]
Thanks!

**Comment by jhecking** [comment]
@aknuds1, thanks for the PR!

The good news is that I have talked to our C client developer, and he will fix the issues with the C client on Alpine Linux, incl. the pthread_attr_setaffinity_np function dependency. (Apparently, there is a different way to set CPU affinity on Alpine Linux.)

The bad news is that we'll probably see a new C client release only next week. Once that's done, I can update the Node.js client to include the fixed C client version.

**Comment by aknuds1** [comment]
Thanks @jhecking for fixing this! I made my own package of the Node.js client in the meantime, so I'm OK in the meantime :)

**Comment by aknuds1** [comment]
Thanks @jhecking!

**Comment by hholst80** [comment]
I ran across this in Alpine 3.15 using Python `torch`

tl;dr I should not waste my time trying to make this work?

```
invoice-serving [master●●] % ./launch.sh             
INFO:     Will watch for changes in these directories: ['/workspace/palette/invoice-serving']
INFO:     Uvicorn running on http://0.0.0.0:8000 (Press CTRL+C to quit)
INFO:     Started reloader process [813] using watchgod
Process SpawnProcess-1:
Traceback (most recent call last):
  File "/usr/lib/python3.9/multiprocessing/process.py", line 315, in _bootstrap
    self.run()
  File "/usr/lib/python3.9/multiprocessing/process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/lib/python3.9/site-packages/uvicorn/subprocess.py", line 76, in subprocess_started
    target(sockets=sockets)
  File "/usr/lib/python3.9/site-packages/uvicorn/server.py", line 60, in run
    return asyncio.run(self.serve(sockets=sockets))
  File "/usr/lib/python3.9/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "uvloop/loop.pyx", line 1501, in uvloop.loop.Loop.run_until_complete
  File "/usr/lib/python3.9/site-packages/uvicorn/server.py", line 67, in serve
    config.load()
  File "/usr/lib/python3.9/site-packages/uvicorn/config.py", line 458, in load
    self.loaded_app = import_from_string(self.app)
  File "/usr/lib/python3.9/site-packages/uvicorn/importer.py", line 21, in import_from_string
    module = importlib.import_module(module_str)
  File "/usr/lib/python3.9/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 850, in exec_module
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
  File "/workspace/palette/invoice-serving/./src/__init__.py", line 1, in <module>
    import torch
  File "/usr/lib/python3.9/site-packages/torch/__init__.py", line 196, in <module>
    _load_global_deps()
  File "/usr/lib/python3.9/site-packages/torch/__init__.py", line 149, in _load_global_deps
    ctypes.CDLL(lib_path, mode=ctypes.RTLD_GLOBAL)
  File "/usr/lib/python3.9/ctypes/__init__.py", line 374, in __init__
    self._handle = _dlopen(self._name, mode)
OSError: Error relocating /usr/lib/python3.9/site-packages/torch/lib/libgomp-a34b3233.so.1: pthread_attr_setaffinity_np: symbol not found
^CINFO:     Stopping reloader process [813]

invoice-serving [master●●] % 

```
